src/app/public/modules/angular-tree/angular-tree-node.component.ts

```diff
@@ -0,0 +1,222 @@
+import {
+  AfterViewInit,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  Input,
+  OnInit,
+  Optional,
+  ViewChild
+} from '@angular/core';
+
+import {
+  ITreeState,
+  TreeNode
+} from 'angular-tree-component';
+
+import {
+  SkyAngularTreeAdapterService
+} from './angular-tree-adapter.service';
+
+import {
+  SkyAngularTreeWrapperComponent
+} from './angular-tree-wrapper.component';
+
+@Component({
+  selector: 'sky-angular-tree-node',
+  templateUrl: './angular-tree-node.component.html'
+})
+export class SkyAngularTreeNodeComponent implements AfterViewInit, OnInit {
+
+  @Input()
+  public index: number;
+
+  @Input()
+  public node: TreeNode;
+
+  @Input()
+  public templates: any;
+
+  public set childFocusIndex(value: number) {
+    if (value !== this._childFocusIndex) {
+      this._childFocusIndex = value;
+      if (this.focusableChildren.length > 0 && value !== undefined) {
+        this.focusableChildren[value].focus();
+      } else {
+        this.nodeContentWrapperRef.nativeElement.focus();
+      }
+    }
+  }
+
+  public get childFocusIndex(): number {
+    return this._childFocusIndex;
+  }
+
+  public set focused(value: boolean) {
+    this.tabIndex = value ? 0 : -1;
+    if (value) {
+      this.nodeContentWrapperRef.nativeElement.focus();
+    }
+  }
+
+  public set isPartiallySelected(value: boolean) {
+    if (value !== this._isPartiallySelected) {
+      this._isPartiallySelected = value;
+      this.changeDetectorRef.markForCheck();
+    }
+  }
+
+  public get isPartiallySelected(): boolean {
+    return this._isPartiallySelected;
+  }
+
+  public set isSelected(value: boolean) {
+    if (value !== this._isSelected) {
+      this._isSelected = value;
+      this.changeDetectorRef.markForCheck();
+    }
+  }
+
+  public get isSelected(): boolean {
+    return this._isSelected;
+  }
+
+  public set tabIndex(value: number) {
+    this._tabIndex = value;
+  }
+
+  public get tabIndex(): number {
+    return this._tabIndex;
+  }
+
+  @ViewChild('nodeContentWrapper')
+  public nodeContentWrapperRef: ElementRef;
+
+  private focusableChildren: HTMLElement[] = [];
+
+  private _childFocusIndex: number;
+
+  private _isPartiallySelected: boolean;
+
+  private _isSelected: boolean;
+
+  private _tabIndex: number = -1;
+
+  constructor(
+    private changeDetectorRef: ChangeDetectorRef,
+    private adapterService: SkyAngularTreeAdapterService,
+    @Optional() private skyAngularTreeWrapper: SkyAngularTreeWrapperComponent
+  ) { }
+
+  public ngOnInit(): void {
+    if (!this.skyAngularTreeWrapper) {
+      console.warn(`<sky-angular-tree-node-wrapper> should be wrapped inside a <sky-angular-tree-wrapper> component.`);
```

Comment:
What if we just threw an error here? Would the component "work" even if the `SkyAngularTreeWrapperComponent` isn't defined?

Reply:
Correct. Done.